src/dbnode/client/fetch_tagged_results_accumulator.go

```diff
@@ -308,13 +328,17 @@ func (accum *fetchTaggedResultAccumulator) AsEncodingSeriesIterators(
 	})
 
 	exhaustive := accum.exhaustive && count <= limit && !moreElems
-	return result, exhaustive, nil
+	return result, FetchResponseMetadata{
+		Exhaustive:         exhaustive,
+		Responses:          len(accum.fetchResponses),
+		EstimateTotalBytes: accum.calcTransport.size,
```

Comment:
A little wary that `calcTransport` can be nil here; may be useful to add in a sanity check here, I think you can do something like this (similar model to [generated protobuf code](https://sourcegraph.com/github.com/m3db/m3/-/blob/src/query/generated/proto/rpcpb/query.pb.go#L226:33)) so that you don't have to add the check everywhere you'd want to access this:

```golang
func (c *calcTransport) GetSize() int {
 if c == nil {
  return 0
 }
 return c.size
}
```